### ISSUE TITLE
`Style/TrailingCommaInArrayLiteral` and `Style/TrailingCommaInHashLiteral` should not apply when terminating brace is on same line.

### ISSUE THREAD

**Issue by armahillo** [rubocop/rubocop#14406]
This is a "bug" in the sense that I believe the cop is not being applied in the spirit of how it was intended, not specifically a syntax or broken code issue. 

## Current Behavior

Violating example that would flag these cops:

```ruby
some_array = ['name',
             'address',
             'phone']
some_hash = { name: 'John',
              address: '123 Street',
              phone: '555-555-5555'}
```
expected "corrected" version:

```ruby
some_array = ['name',
             'address',
             'phone',]         # terminal comma included
some_hash = { name: 'John',
              address: '123 Street',
              phone: '555-555-5555',} # terminal comma included
```

My understanding of the purpose of this cop is to reduce line churn when a new line is added, eg.

```ruby
some_array = ['name',
             'address',
             'phone']
# to
some_array = ['name',
             'address',
             'phone',  # this line appears in the diff also for having the comma added
             'email']
```

Except that when the terminating brace is on the same line, you're still going to have that additional line included:

```ruby
some_array = ['name',
             'address',
             'phone',]
# to
some_array = ['name',
             'address',
             'phone',  # Still changed, because the terminating brace was here previously
             'email',]
```

## Desired change

1. If the line ends with a terminating brace, the trailing comma is not required.
2. If the line does not end with a terminating brace, the trailing comma is required

```ruby
# Good
some_array = ['name',
             'address',
             'phone']
some_array = ['name',
             'address',
             'phone',
             ]
# Bad
some_array = ['name',
             'address',
             'phone',]
some_array = ['name',
             'address',
             'phone'
             ]
```

**Comment by issyl0** [comment]
Using the [`EnforcedStyleForMultiline: diff_comma`](https://docs.rubocop.org/rubocop/cops_style.html#enforcedstyleformultiline_-diff_comma-styletrailingcommainhashliteral) configuration should get the behaviour you intend here!

**Comment by bbatsov** [comment]
This is already supported via `EnforcedStyleForMultiline: diff_comma` (as @issyl0 mentioned above) - it requires a trailing comma only when the last item precedes a newline, not when the closing brace is on the same line. Closing as already covered.
